File: app/src/main/java/com/vegasoft/mypasswords/data/db/BaseDBHelper.java
package com.vegasoft.mypasswords.data.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

class BaseDBHelper extends SQLiteOpenHelper {

    // Table names
    static final String TABLE_RECORD = "records_table";

    //Record table
    static final String RECORD_ID = "id";
    static final String RECORD_NAME = "name";
    static final String RECORD_DATE = "date";
    static final String RECORD_GROUP = "group_name";
    static final String RECORD_SITE = "site_url";
    static final String RECORD_USER = "user";
    static final String RECORD_PASS = "pass";
    static final String RECORD_IMAGE = "image_path";
    static final String RECORD_ENCRYPTION = "encryption";

    // All Static variables
    private static final String TAG = BaseDBHelper.class.getSimpleName();
    // Database Version
    private static final int DATABASE_VERSION = 1;
    // Database Name
    private static final String DATABASE_NAME = "PreferenceCentral.db";

    BaseDBHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        createRecordsTable(db);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        dropRecordsTable(db);
        onCreate(db);
    }

    @Override
    public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        onUpgrade(db, oldVersion, newVersion);
    }

    private void createRecordsTable(SQLiteDatabase db) {
        if (db == null)
            db = getWritableDatabase();
        db.execSQL("CREATE TABLE " + TABLE_RECORD + "("
                + RECORD_ID + " LONG PRIMARY KEY, "
                + RECORD_NAME + " TEXT, "
                + RECORD_DATE + " LONG, "
                + RECORD_GROUP + " TEXT, "
                + RECORD_SITE + " TEXT, "
                + RECORD_USER + " TEXT, "
                + RECORD_PASS + " TEXT, "
                + RECORD_IMAGE + " TEXT, "
                + RECORD_ENCRYPTION + " INTEGER ) " );
        Log.d(TAG, "SQLiteDatabase onCreate(SQLiteDatabase db) "
                + TABLE_RECORD);
    }

    private void dropRecordsTable(SQLiteDatabase db) {
        if (db == null)
            db = getWritableDatabase();
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_RECORD);
    }
}

File: app/src/main/java/com/vegasoft/mypasswords/data/db/BaseDBHelper.kt
package com.vegasoft.mypasswords.data.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.util.Log

open class BaseDBHelper(context: Context?) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
    override fun onCreate(db: SQLiteDatabase) {
        createRecordsTable(db)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        dropRecordsTable(db)
        onCreate(db)
    }

    override fun onDowngrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        onUpgrade(db, oldVersion, newVersion)
    }

    private fun createRecordsTable(_db: SQLiteDatabase) {
        val db: SQLiteDatabase? = if (_db == null) writableDatabase else _db
        db!!.execSQL("CREATE TABLE " + TABLE_RECORD + "("
                + RECORD_ID + " LONG PRIMARY KEY, "
                + RECORD_NAME + " TEXT, "
                + RECORD_DATE + " LONG, "
                + RECORD_GROUP + " TEXT, "
                + RECORD_SITE + " TEXT, "
                + RECORD_USER + " TEXT, "
                + RECORD_PASS + " TEXT, "
                + RECORD_IMAGE + " TEXT, "
                + RECORD_ENCRYPTION + " INTEGER ) ")
        Log.d(TAG, "SQLiteDatabase onCreate(SQLiteDatabase db) "
                + TABLE_RECORD)
    }

    private fun dropRecordsTable(_db: SQLiteDatabase) {
        val db: SQLiteDatabase? = if (_db == null) writableDatabase else _db
        db?.execSQL("DROP TABLE IF EXISTS $TABLE_RECORD")
    }

    companion object {
        // Table names
        const val TABLE_RECORD = "records_table"

        //UIRecord table
        const val RECORD_ID = "id"
        const val RECORD_NAME = "name"
        const val RECORD_DATE = "date"
        const val RECORD_GROUP = "group_name"
        const val RECORD_SITE = "site_url"
        const val RECORD_USER = "user"
        const val RECORD_PASS = "pass"
        const val RECORD_IMAGE = "image_path"
        const val RECORD_ENCRYPTION = "encryption"

        // All Static variables
        private val TAG = BaseDBHelper::class.java.simpleName

        // Database Version
        private const val DATABASE_VERSION = 2

        // Database Name
        private const val DATABASE_NAME = "PSM.db"
    }
}